thefuck/rules/git_push.py

```diff
@@ -6,7 +6,7 @@
 @git_support
 def match(command):
     return ('push' in command.script_parts
-            and 'set-upstream' in command.output)
+            and 'git push --set-upstream' in command.output)
```

Comment:
I'm thinking we should just make this line be:

```python
            and '--set-upstream' in command.script_parts)
```

in order to allow `--set-upstream` to occur elsewhere than right after `push`. Unless someone names a branch `--set-upstream`, this should be fine.